src/bin/wd_copier.rs

```diff
@@ -12,10 +15,40 @@ struct Opt {
     dst: PathBuf,
 }
 
+fn get_dirty_bytes() -> u64 {
+    let meminfo = procfs::Meminfo::new().unwrap();
+    meminfo.dirty
+}
+
+fn sync_progress_bar(
+    rx: mpsc::Receiver<()>,
+    mut progress_bar: progress::Bar,
+    starting_dirty: u64,
+    dirty_after_copy: u64,
+) {
+    progress_bar.set_job_title("syncing... (2/2)");
+    loop {
+        let percent = 100
+            - ((get_dirty_bytes().saturating_sub(starting_dirty))
+                / (dirty_after_copy / 100));
```

Comment:
This expression is a bit complicated to reason about. Maybe add a new `calc_percentage` function that takes a min-max range and the current value.

Reply:
b63e642